### PR TITLE
Enable LayoutRenderer to ResolveService during intialization

### PR DESCRIPTION
**Pull request by snakefoot**
Also avoid throwing exceptions during initialization, when dependency is available from external service-provider (NLog should attempt to be silent partner).

### CODE REVIEW

File: src/NLog/Config/ServiceRepository.cs [hunk 1]
```diff
@@ -53,6 +53,8 @@ public abstract class ServiceRepository : IServiceProvider
         /// <remarks>Avoid calling this while handling a LogEvent, since random deadlocks can occur.</remarks>
         public abstract object GetService(Type serviceType);
 
+        internal abstract object TryGetService(Type serviceType);
```
Comment (304NotModified) [inline src/NLog/Config/ServiceRepository.cs:56]: With A Try, I expected a bool resturn and a out param.

Proposal: rename to FindService

Comment (304NotModified) [reply src/NLog/Config/ServiceRepository.cs:56]: it's a common naming convention for me (find returns `null` when it's not there, get thows an exception for that case).

 See also https://softwareengineering.stackexchange.com/a/182120/16953

File: src/NLog/Config/ServiceRepositoryExtensions.cs [hunk 2]
```diff
@@ -47,30 +47,50 @@ internal static ServiceRepository GetServiceProvider([CanBeNull] this LoggingCon
...
-            catch (NLogDependencyResolveException)
+            else
             {
-                if (ignoreExternalProvider)
-                    throw;
+                IServiceProvider externalServiceProvider;
+
+                try
+                {
+                    var service = serviceProvider.TryGetService(typeof(T)) as T;
```
Comment (304NotModified) [inline src/NLog/Config/ServiceRepositoryExtensions.cs:60]: generic method next, same as GetService, would be nice. 

(consistent API)

**Review by 304NotModified — COMMENTED** [review]
Looks good, some small notes

File: src/NLog/Config/ServiceRepository.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/Config/ServiceRepository.cs]: Yes thank you for spotting. Fixed TryGetService so it returns true / false

File: src/NLog/Config/ServiceRepositoryExtensions.cs (hunk 2 above)
Comment (snakefoot) [reply src/NLog/Config/ServiceRepositoryExtensions.cs]: Yes thank you for spotting. Fixed TryGetService to be generic method.

**Pull request merged**